homeassistant/components/volumio/media_player.py

```diff
@@ -68,91 +53,59 @@
 
 PLAYLIST_UPDATE_INTERVAL = timedelta(seconds=15)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    }
-)
-
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Volumio platform."""
-    if DATA_VOLUMIO not in hass.data:
-        hass.data[DATA_VOLUMIO] = {}
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Volumio media player platform."""
 
-    # This is a manual configuration?
-    if discovery_info is None:
-        name = config.get(CONF_NAME)
-        host = config.get(CONF_HOST)
-        port = config.get(CONF_PORT)
-    else:
-        name = "{} ({})".format(DEFAULT_NAME, discovery_info.get("hostname"))
-        host = discovery_info.get("host")
-        port = discovery_info.get("port")
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    volumio = data[DATA_VOLUMIO]
+    info = data[DATA_INFO]
+    uid = config_entry.data[CONF_ID]
+    name = config_entry.data[CONF_NAME]
 
-    # Only add a device once, so discovered devices do not override manual
-    # config.
-    ip_addr = socket.gethostbyname(host)
-    if ip_addr in hass.data[DATA_VOLUMIO]:
-        return
-
-    entity = Volumio(name, host, port, hass)
-
-    hass.data[DATA_VOLUMIO][ip_addr] = entity
+    entity = Volumio(hass, volumio, uid, name, info)
     async_add_entities([entity])
 
 
 class Volumio(MediaPlayerEntity):
     """Volumio Player Object."""
 
-    def __init__(self, name, host, port, hass):
+    def __init__(self, hass, volumio, uid, name, info):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.